Feature/replace npm package with api

Rewanos opened the pull request.

boale committed:
feat: replace quotesy with API call

add env config
add API service
update app workflow
add/update unit tests
exclude *.mock.ts from coverage
update deps

boale committed:
build: improve pipeline build

upd test api endpoint

boale committed:
Merge remote-tracking branch 'origin/develop' into feature/replace-npm-package-with-api

boale committed:
Merge branch 'develop' into feature/replace-npm-package-with-api

boale committed:
build: improve pipeline build

update envs' API endpoints

boale committed:
build: improve pipeline build

update codecov upload command

boale committed:
build: improve pipeline build

update codecov upload command

boale committed:
build: improve pipeline build

update codecov upload command

boale committed:
build: improve pipeline build

get rid of codecov in pipeline

boale committed:
build: improve pipeline build

get rid of codecov in pipeline

boale committed:
build: improve pipeline build

get rid of codecov in pipeline

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/Rewanos/ngQuote/c864acbdde7151766cf7fc5f86a7a4769e3f8869/Rewanos/ngQuote/62b63bd9412eecc98daf4ada508dabee8e82f5fa/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0) your code in 1.209 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

##### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/Rewanos/ngQuote/c864acbdde7151766cf7fc5f86a7a4769e3f8869/Rewanos/ngQuote/62b63bd9412eecc98daf4ada508dabee8e82f5fa/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0)
---

boale committed:
build: improve pipeline (#33)

* build: improve pipeline

- add maintenance page
- add/upd buildspecs
- upd CF template
- upd scripts
- clean up
- bump quotesy version
- fix/style template

* chore: bump angular version

boale committed:
Merge remote-tracking branch 'origin/develop' into feature/replace-npm-package-with-api

# Conflicts:
#	buildspec/buildspec-quality.yml
#	buildspec/buildspec.prod.yml
#	buildspec/buildspec.test.yml
#	package-lock.json
#	package.json
#	src/app/quote/services/quote.service.spec.ts

ghost commented:
## DeepCode's analysis on [#fceaf5](https://www.deepcode.ai/app/gh/Rewanos/ngQuote/441300b5441de24cfcdfc1147feb0052d72b9324/Rewanos/ngQuote/fceaf53275cd53d5dcce3177a5c325d73480ab11/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) found:

- :information_source: **1** minor issue. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type. Occurrences: <ul><li><a href="https://github.com/Rewanos/ngQuote/blob/fceaf53275cd53d5dcce3177a5c325d73480ab11/src/app/quote/services/quote.service.spec.ts#L19">quote.service.spec.ts:19</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/Rewanos/ngQuote/441300b5441de24cfcdfc1147feb0052d72b9324/Rewanos/ngQuote/fceaf53275cd53d5dcce3177a5c325d73480ab11/pr/_/%2Fsrc%2Fapp%2Fquote%2Fservices%2Fquote.service.spec.ts/javascript%2Ftslint%2Fno-any/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/Rewanos/ngQuote/441300b5441de24cfcdfc1147feb0052d72b9324/Rewanos/ngQuote/fceaf53275cd53d5dcce3177a5c325d73480ab11/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=Rewanos)
